packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts

```diff
@@ -186,34 +186,35 @@ export class BeaconWallet implements WalletProvider {
     await this.client.setActiveAccount();
   }
 
-  async sign(signingRequest: { payload: string; watermark?: Uint8Array }) {
-    let bb = hex2buf(signingRequest.payload);
-    if (typeof signingRequest.watermark !== 'undefined') {
-      bb = mergebuf(signingRequest.watermark, bb);
+  async sign({ payload, watermark }: { payload: string; watermark?: Uint8Array }) {
```

Comment:
besides this file `packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts` and `packages/taquito/src/wallet/interface.ts`
also have define sign function that shall be updated along with the change